### PR TITLE
feat: Link to rescheduled booking from the old one

### LINKED ISSUE #14707: [CAL-3755] Link to rescheduled booking from the old one
### Is your proposal related to a problem?

Rescheduling a booking turns the old booking "Reschedule"/"Cancel" links into dead links. The email for the original is still in users' inboxes and they might click on the wrong one by mistake.

### Describe the solution you'd like

Edit the "Notes" of the old booking to add a link to the new, rescheduled one.

### Describe alternatives you've considered

Hope that attendees never click on the wrong email.

The link could be a first-class field, rather than shoving it into "Additional notes". But additional notes just seems like an easy way to do it.

### Additional context

This is the flow which highlighted this for me:

- Book a meeting from my usual page
- Click the "Reschedule" link in the email confirmation
- Reschedule it to another time
- Go back to my email inbox, and click "Reschedule" again on the original email

I then see a "This meeting has been cancelled" screen with no hint anywhere that it's actually rescheduled.

This could also happen if we share a cal.com link with users of our platform (we have a Team account and are (planning to) use cal.com to let our users schedule bookings).

Note: the Google Meet link seems to be unchanged, so that's good/mitigates this to some extent.

### Requirement/Document

?

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

<sub>[CAL-3755](https://linear.app/calcom/issue/CAL-3755/link-to-rescheduled-booking-from-the-old-one)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

- Fixes #14707 
- Fixes CAL-3755

Screenshot 2024-07-16 204910

https://github.com/calcom/cal.com/assets/74371312/84df993b-a56b-4e69-9c5c-a90e3316dae9

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

**Comment by Amit91848** [comment]
Currently just adds it to notes, lmk if you want to have it as first class field

**Review by ibex088 — DISMISSED** [review]
Tested, LGTM!!

**Review by keithwillcode — DISMISSED** [review]
Would like sign off from @ciaranha / @PeerRich before we merge this.

**Comment by hariombalhara** [comment]
I don't prefer updating the description on cancellation. `description` is a property of the booking that could have some existing value that gets overwritten in this way.

Why don't we simply use `fromReschedule` to identify the booking to which the cancelled booking has been scheduled. It should allow us to show the value on the booking details page.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
I don't prefer updating the description on cancellation. `description` is a property of the booking that could have some existing value that gets overwritten in this way.

Why don't we simply use `fromReschedule` to identify the booking to which the cancelled booking has been scheduled. It should allow us to show the value on the booking details page.

**Comment by Amit91848** [comment]
> I don't prefer updating the description on cancellation. `description` is a property of the booking that could have some existing value that gets overwritten in this way.

Previous description is still present it just adds a line to it, but I get your point.

> Why don't we simply use `fromReschedule` to identify the booking to which the cancelled booking has been scheduled. It should allow us to show the value on the booking details page.

Will make changes to it

**Comment by Amit91848** [comment]
Made changes @hariombalhara , please review again


https://github.com/calcom/cal.com/assets/74371312/84df993b-a56b-4e69-9c5c-a90e3316dae9

File: apps/web/modules/bookings/views/bookings-single-view.tsx [hunk 1]
```diff
@@ -585,6 +585,22 @@ export default function Success(props: PageProps) {
...
+                                <Link href={`/booking/${rescheduledToUid}`}>
+                                  <div className="flex items-center gap-1">
+                                    {t("Link")}
+                                    <Icon name="external-link" className="h-4 w-4" />
+                                  </div>
+                                </Link>
+                              </span>
+                            </div>
+                          </>
+                        )}
```
Comment (hariombalhara) [inline apps/web/modules/bookings/views/bookings-single-view.tsx:602]: Let's move it out into it's own component within this module.

Possible name could be <RescheduledToLink>

```
{rescheduledToUid ? <RescheduledToLink>: null}
```

File: packages/features/bookings/lib/getRescheduledToBooking.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const getRescheduledToBooking = async (originalBookingUid: string) => {
+  const rescheduledTo = await prisma.booking.findFirst({
+    where: {
+      fromReschedule: originalBookingUid,
+    },
+    select: {
+      uid: true,
+    },
+  });
```
[the remark this answers is not in the document]
Comment (Amit91848) [reply packages/features/bookings/lib/getRescheduledToBooking.tsx:11]: Done

**Comment by Amit91848** [comment]
> @Amit91848 Could you add the updated snapshot in the PR description.
> 
> We are also waiting for @ciaranha @PeerRich to approve it as well

Updated the description

**Review by hariombalhara — DISMISSED** [review]
Looks good !!

@PeerRich @ciaranha Could you confirm the design here? Should be quick.

**Comment by ciaranha** [comment]
Can we just edit the copy 

Change "Rescheduled to" to "Rescheduled". "Link to" to "View booking"

**Comment by Amit91848** [comment]
> Can we just edit the copy

Not sure wdym by that

> Change "Rescheduled to" to "Rescheduled". "Link to" to "View booking"

Changed the text @ciaranha

**Comment by ciaranha** [comment]
> Not sure wdym by that

I just meant to update the text like you did :) (Update copy = Update text)

Thanks!

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**